File: Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root == None:
            return 0
        if root.left == None and root.right != None:
            return self.minDepth(root.right) + 1
        if root.left != None and root.right == None:
            return  self.minDepth(root.left) + 1
        return min(self.minDepth(root.left), self.minDepth(root.right)) + 1




a = Solution()

File: Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.swift
import Foundation

/*
 Given a binary tree, find its minimum depth.

 The minimum depth is the number of nodes along the shortest path from the root node down to the nearest leaf node.
 */

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */
class Solution {

    class TreeNode {
        var val: Int
        var left: TreeNode?
        var right: TreeNode?

        init(_ val: Int) {
            self.val = val
            self.left = nil
            self.right = nil
        }
    }

    func minDepth(_ root: TreeNode?) -> Int {
        if root == nil {
            return 0
        }
        if root!.left != nil && root!.right == nil {
            return minDepth(root!.left) + 1
        }

        if root!.left == nil && root!.right != nil {
            return minDepth(root!.right) + 1
        }

        return min(minDepth(root!.left), minDepth(root!.right)) + 1
    }
}





